bindgen/ir/ty.rs

```diff
@@ -1099,6 +1100,20 @@ impl Type {
                         // within the clang parsing.
                         TypeKind::Opaque
                     } else {
+                        // Check if this type definition is an alias to a pointer of a `const
+                        // struct` with the same name and add the `_ptr` suffix to it to avoid name
+                        // collisions.
+                        if !ctx.options().c_naming {
+                            if let Some(inner_name) = dbg!(inner_spelling
+                                .strip_prefix("const struct ")
+                                .and_then(|s| s.strip_suffix(" *")))
```

Comment:
This textual approach seems unreliable; what if there isn't a space before the `*`? Possibly other cases?

Reply:
This textual representation is generated by `clang` and it doesn't seem to be affected by adding/removing spaces from the source code. For example:
```
typedef const struct foo {
    char bar;
}*foo;
```
works just fine.